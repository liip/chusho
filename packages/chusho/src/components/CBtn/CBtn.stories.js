import StoryRouter from 'storybook-vue-router';
import CBtn from './CBtn';

export default {
  title: 'Components|Btn',
  component: CBtn,
  parameters: {
    componentSubtitle:
      'Uniformized button style for `router-link`, `nuxt-link`, `a` or `button` elements.',
    options: {
      componentConfig: [
        {
          name: 'defaultClass',
          type: { summary: 'string' },
          description: 'Class applied to all CBtn components.',
        },
        {
          name: 'disabledClass',
          type: { summary: 'string' },
          description:
            'Class applied to Btn when prop `disabled` is set to `true`.',
        },
        {
          name: 'variants',
          type: { summary: 'object' },
          description:
            'Predefined set of variants.\n\nFor example: `{ "primary": "bg-blue-500 text-white" }`',
        },
      ],
    },
  },
  decorators: [
    StoryRouter(
      {},
      {
        routes: [
          {
            path: '/',
            name: 'home',
            component: { template: `<div>Home</div>` },
          },
        ],
      }
    ),
  ],
};

export const Default = () => ({
  components: { CBtn },
  template: '<CBtn>Click me</CBtn>',
});

export const AsLink = () => ({
  components: { CBtn },
  template: '<CBtn href="#">Click me</CBtn>',
});

export const AsRouterLink = () => ({
  components: { CBtn },
  template: `<CBtn :to="{ name: 'home' }">Click me</CBtn>`,
});

export const TypeSubmit = () => ({
  components: { CBtn },
  template: '<CBtn type="submit">Click me</CBtn>',
});

export const WithVariant = () => ({
  components: { CBtn },
  template: '<CBtn variant="default medium">Click me</CBtn>',
});

export const Disabled = () => ({
  components: { CBtn },
  template: `<CBtn disabled>Click me</CBtn>`,
});

export const WithCustomClasses = () => ({
  components: { CBtn },
  template: `<CBtn class="font-bold">Click me</CBtn>`,
});
