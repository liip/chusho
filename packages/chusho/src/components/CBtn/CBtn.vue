<script lang="ts">
import Vue, { VNodeData } from 'vue';
import { mergeData } from 'vue-functional-data-merge';
import { VNode } from 'vue/types/umd';

interface BtnProps {
  href: string;
  type: string;
  variant: string;
  disabled: boolean;
}

export default Vue.extend<BtnProps>({
  name: 'CBtn',

  functional: true,

  props: {
    /**
     * Make the button a link to the given location
     */
    href: {
      type: String,
      default: null,
    },

    /**
     * In case the button is a `button` element, specify its type
     */
    type: {
      type: String,
      default: 'button',
    },

    /**
     * Customize the button appearance by applying one or multiple variants defined in the config (btn.variants).
     * Example: `"primary large"`
     */
    variant: {
      type: String,
      default: '',
    },

    /**
     * Disable the button (doesnâ€™t apply to links) and apply the associated class from the config (btn.disabled).
     */
    disabled: {
      type: Boolean,
      default: false,
    },
  },

  render(h, { props, data, parent, children }): VNode {
    const { btn } = parent.$chusho.options;
    const tag = props.href ? 'a' : 'button';

    const classes = [btn.default, { [btn.disabled]: props.disabled }];
    const variants = props.variant.split(' ');
    variants.forEach(variant => {
      const target = btn.variants[variant];
      if (target) classes.push(target);
    });

    const componentData: VNodeData = {
      class: classes,
      attrs: {
        href: props.href,
        // Concerns only on button tags, skip for anchors
        ...(props.disabled && tag === 'button' && { disabled: true }),
        ...(props.type && tag === 'button' && { type: props.type }),
      },
    };

    return h(tag, mergeData(data, componentData), children);
  },
});
</script>
