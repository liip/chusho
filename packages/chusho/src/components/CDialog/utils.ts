export const PORTAL_ID = 'chusho-dialogs-portal';

export function createPortalIfNotExists(): void {
  if (!document.getElementById(PORTAL_ID)) {
    const portalNode = document.createElement('div');
    portalNode.setAttribute('id', PORTAL_ID);
    document.body.insertAdjacentElement('beforeend', portalNode);
  }
}

export function getPortalNodeSiblings(): HTMLElement[] {
  return Array.from(
    document.body.children as HTMLCollectionOf<HTMLElement>
  ).filter((el) => {
    // Ignore already non-visible elements such as <script>
    return el.offsetParent !== null && el.id !== PORTAL_ID;
  });
}

export function preventAccessToPageContent(): void {
  getPortalNodeSiblings().forEach((el) => {
    el.setAttribute('aria-hidden', 'true');
  });
}

export function releaseAccessToPageContent(): void {
  getPortalNodeSiblings().forEach((el) => {
    el.removeAttribute('aria-hidden');
  });
}
