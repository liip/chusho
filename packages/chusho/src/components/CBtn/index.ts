import CBtn from './CBtn.vue';

export { CBtn };
export default CBtn;
