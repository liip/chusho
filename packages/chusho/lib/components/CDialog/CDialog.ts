import {
  PropType,
  Teleport,
  VNode,
  defineComponent,
  h,
  inject,
  mergeProps,
  nextTick,
} from 'vue';

import { DollarChusho } from '../../types';

import componentMixin from '../mixins/componentMixin';
import transitionMixin from '../mixins/transitionMixin';

import useActiveElement from '../../composables/useActiveElement';
import useComponentConfig from '../../composables/useComponentConfig';

import {
  generateConfigClass,
  renderWithTransition,
} from '../../utils/components';
import { getFocusableElements } from '../../utils/keyboard';
import { isClient, isServer } from '../../utils/ssr';
import {
  PORTAL_ID,
  createPortalIfNotExists,
  preventAccessToPageContent,
  releaseAccessToPageContent,
} from './utils';

export interface DialogData {
  active: boolean;
  focusableElements: Array<HTMLElement>;
}

export default defineComponent({
  name: 'CDialog',

  mixins: [componentMixin, transitionMixin],

  inheritAttrs: false,

  props: {
    /**
     * Define if the dialog should be visible or not
     */
    modelValue: {
      type: Boolean,
      required: true,
    },
    /**
     * Attributes to be applied to the overlay element.
     *
     * For example: `{ class: 'custom-overlay', id: 'gallery-modal'  }`
     */
    overlay: {
      type: Object as PropType<Record<string, unknown>>,
      default: () => ({}),
    },
  },

  emits: ['update:modelValue'],

  setup() {
    const chusho = inject<DollarChusho | null>('$chusho', null);

    return {
      config: useComponentConfig('dialog'),
      chusho,
      activeElement: useActiveElement(),
    };
  },

  data(): DialogData {
    return {
      active: false,
      focusableElements: [],
    };
  },

  computed: {
    firstFocusableEl(): HTMLElement | undefined {
      return this.focusableElements[0];
    },

    lastFocusableEl(): HTMLElement | undefined {
      return this.focusableElements[this.focusableElements.length - 1];
    },
  },

  updated() {
    if (this.modelValue && this.active) {
      this.refreshFocusableElements();
    }
  },

  beforeUnmount() {
    document.removeEventListener('keydown', this.handleKeyDown);
  },

  methods: {
    refreshFocusableElements(): void {
      if (this.$refs.dialogElement && this.modelValue) {
        this.focusableElements = getFocusableElements(
          this.$refs.dialogElement as HTMLElement
        );
      }
    },

    activate(): void {
      if (this.active || isServer) return;

      this.active = true;
      this.activeElement.save();

      // Store the dialog instance in a global array
      // This allow closing only the last one when pressing ESC and multiple dialogs are open
      this.chusho?.openDialogs.push(this);

      this.refreshFocusableElements();
      this.firstFocusableEl?.focus();
      preventAccessToPageContent();

      document.addEventListener('keydown', this.handleKeyDown);
    },

    deactivate(): void {
      if (!this.active || isServer) return;

      releaseAccessToPageContent();

      const index = this.chusho?.openDialogs.indexOf(this);
      if (index) {
        this.chusho?.openDialogs.splice(index, 1);
      }

      this.activeElement.restore();
      document.removeEventListener('keydown', this.handleKeyDown);
      this.active = false;
    },

    handleKeyDown(e: KeyboardEvent): void {
      switch (e.key) {
        case 'Esc':
        case 'Escape':
          if (
            this ===
            this.chusho?.openDialogs[this.chusho.openDialogs.length - 1]
          ) {
            this.$emit('update:modelValue', false);
          }
          break;

        case 'Tab':
          if (this.focusableElements.length === 1) {
            e.preventDefault();
          } else if (e.shiftKey) {
            if (
              this.lastFocusableEl &&
              document.activeElement === this.firstFocusableEl
            ) {
              e.preventDefault();
              this.lastFocusableEl.focus();
            }
          } else {
            if (
              this.firstFocusableEl &&
              document.activeElement === this.lastFocusableEl
            ) {
              e.preventDefault();
              this.firstFocusableEl.focus();
            }
          }
          break;
      }
    },

    renderChildren() {
      const children: VNode[] = [];

      if (this.modelValue) {
        const overlayProps: Record<string, unknown> = mergeProps(this.overlay, {
          tabindex: '-1',
          onClick: (e: KeyboardEvent) => {
            if (e.target !== e.currentTarget) return;
            this.$emit('update:modelValue', false);
          },
          ...generateConfigClass(this.config?.overlayClass, this.$props),
        });

        const dialogProps = mergeProps(this.$attrs, {
          role: 'dialog',
          ref: 'dialogElement',
          ...generateConfigClass(this.config?.class, this.$props),
        });

        children.push(
          h('div', overlayProps, h('div', dialogProps, this.$slots))
        );

        nextTick(this.activate);
      } else {
        nextTick(this.deactivate);
      }

      return children;
    },
  },

  render() {
    if (isClient) {
      createPortalIfNotExists();
    }

    return h(
      Teleport,
      {
        to: `#${PORTAL_ID}`,
      },
      [
        renderWithTransition(
          this.renderChildren,
          this.transition,
          this.config?.transition
        ),
      ]
    );
  },
});
