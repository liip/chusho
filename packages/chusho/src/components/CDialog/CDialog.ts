import Vue, { VNode, VNodeData } from 'vue';
import { mergeData } from 'vue-functional-data-merge';
import { MountingPortal } from 'portal-vue';
import ClientOnly from 'vue-client-only';
import { filterVueData, isPlainObject } from '../../utils/objects';
import { getFocusableElements } from '../../utils/keyboard';

const PORTAL_ID = 'chusho-dialogs-portal';
const KEY_TAB = 9;
const KEY_ESC = 27;

interface DialogData {
  savedActiveElement: null | HTMLElement;
  focusableElements: Array<HTMLElement>;
}

export default Vue.extend({
  name: 'CDialog',

  model: {
    prop: 'open',
    event: 'toggle',
  },

  props: {
    /**
     * Control the Dialog state
     */
    open: {
      type: Boolean,
      required: true,
    },
    /**
     * The object can contain any Vue built-in [transition component props](https://vuejs.org/v2/api/#transition).
     *
     * For example: `{ name: "fade", mode: "out-in" }`.
     *
     * If you defined a default transition in the config and want to disable it, use `false`.
     */
    transition: {
      type: [Object, Boolean],
      default: null,
    },
  },

  data(): DialogData {
    return {
      savedActiveElement: null,
      focusableElements: [],
    };
  },

  computed: {
    firstFocusableEl(): HTMLElement | undefined {
      return this.focusableElements[0];
    },

    lastFocusableEl(): HTMLElement | undefined {
      return this.focusableElements[this.focusableElements.length - 1];
    },
  },

  methods: {
    activate(): void {
      if (this.$chusho.openDialogs.includes(this)) return;

      this.savedActiveElement = document.activeElement as HTMLElement;

      // Store the dialog instance in a global array
      // This allow closing only the last one when pressing ESC and multiple dialogs are open
      this.$chusho.openDialogs.push(this);

      this.$nextTick(() => {
        this.focusableElements = getFocusableElements(
          this.$refs.dialog as Element
        );
        this.firstFocusableEl?.focus();
        this.preventAccessToPageContent();
      });

      document.addEventListener('keydown', this.handleKeyDown);
    },

    deactivate(): void {
      this.releaseAccessToPageContent();

      const i = this.$chusho.openDialogs.indexOf(this);
      this.$chusho.openDialogs.splice(i, 1);

      this.savedActiveElement?.focus();

      document.removeEventListener('keydown', this.handleKeyDown);
    },

    createPortalIfNotExists(): void {
      if (!document.getElementById(PORTAL_ID)) {
        const portalNode = document.createElement('div');
        portalNode.setAttribute('id', PORTAL_ID);
        document.body.insertAdjacentElement('beforeend', portalNode);
      }
    },

    getPortalNodeSiblings(): HTMLElement[] {
      return Array.from(
        document.body.children as HTMLCollectionOf<HTMLElement>
      ).filter((el) => {
        // Ignore already non-visible elements such as <script>
        return el.offsetParent !== null && el.id != PORTAL_ID;
      });
    },

    preventAccessToPageContent(): void {
      this.getPortalNodeSiblings().forEach((el) => {
        el.setAttribute('aria-hidden', 'true');
      });
    },

    releaseAccessToPageContent(): void {
      this.getPortalNodeSiblings().forEach((el) => {
        el.removeAttribute('aria-hidden');
      });
    },

    handleKeyDown(e: KeyboardEvent): void {
      switch (e.keyCode) {
        case KEY_ESC:
          if (
            this ===
            this.$chusho.openDialogs[this.$chusho.openDialogs.length - 1]
          ) {
            this.$emit('toggle', false);
          }
          break;

        case KEY_TAB:
          if (this.focusableElements.length === 1) {
            e.preventDefault();
          } else if (e.shiftKey) {
            if (
              this.lastFocusableEl &&
              document.activeElement === this.firstFocusableEl
            ) {
              e.preventDefault();
              this.lastFocusableEl.focus();
            }
          } else {
            if (
              this.firstFocusableEl &&
              document.activeElement === this.lastFocusableEl
            ) {
              e.preventDefault();
              this.firstFocusableEl.focus();
            }
          }
          break;
      }
    },
  },

  render(h): VNode {
    const dialogConfig = this.$parent?.$chusho?.options?.components?.dialog;
    let children: VNode[] = [];

    // Do not render dialogs on the server
    if (!this.$isServer) {
      this.createPortalIfNotExists();

      if (this.open) {
        const overlayData: VNodeData = {
          attrs: {
            tabindex: '-1',
          },
          on: {
            click: (e: KeyboardEvent) => {
              if (e.target !== e.currentTarget) return;
              this.$emit('toggle', false);
            },
          },
        };
        const dialogData: VNodeData = {
          attrs: {
            role: 'dialog',
          },
          ref: 'dialog',
        };

        if (dialogConfig?.overlayClass) {
          overlayData.class = dialogConfig.overlayClass;
        }
        if (dialogConfig?.dialogClass) {
          dialogData.class = dialogConfig.dialogClass;
        }

        children.push(
          h('div', overlayData, [
            h(
              'div',
              mergeData(filterVueData(this.$vnode.data), dialogData),
              this.$slots.default
            ),
          ])
        );

        this.$nextTick(this.activate);
      } else {
        this.$nextTick(this.deactivate);
      }

      let transition;

      if (isPlainObject(this.transition)) {
        transition = this.transition;
      } else if (
        this.transition !== false &&
        dialogConfig &&
        dialogConfig.transition
      ) {
        transition = dialogConfig.transition;
      }

      if (transition) {
        children = [
          h(
            'transition',
            {
              props: transition,
              key: 'transition',
            },
            children
          ),
        ];
      }
    }

    return h(ClientOnly, [
      h(
        MountingPortal,
        {
          props: {
            append: true,
            mountTo: `#${PORTAL_ID}`,
          },
        },
        children
      ),
    ]);
  },
});
