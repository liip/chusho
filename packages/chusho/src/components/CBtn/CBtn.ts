import {
  defineComponent,
  h,
  inject,
  mergeProps,
  PropType,
  resolveComponent,
} from 'vue';
import { RouteLocationRaw } from 'vue-router';

import { DollarChusho } from '../../types';
import { generateConfigClass } from '../../utils/components';
import { warn } from '../../utils/debug';
import componentMixin from '../mixin';

export default defineComponent({
  name: 'CBtn',

  mixins: [componentMixin],

  inheritAttrs: false,

  props: {
    /**
     * Make the button a link to the given URL.
     */
    href: {
      type: String,
      default: null,
    },

    /**
     * Use a `router-link` (or `nuxt-link` in a Nuxt app) in the background, works juste like the `router-link` prop of the same name.
     *
     * Note that you can also pass all other `router-link` props such as `activeClass`, `replace`, … See [Vue router documentation](https://router.vuejs.org/api/#router-link) for a complete list of available props.
     */
    to: {
      type: [String, Object] as PropType<RouteLocationRaw>,
      default: null,
    },

    /**
     * Specifies the button type, does not apply when `to` or `href` props are used. Example: `submit`.
     */
    type: {
      type: String,
      default: 'button',
    },

    /**
     * Disable the button, does not apply when `to` or `href` props are used.
     */
    disabled: {
      type: Boolean,
      default: false,
    },
  },

  render() {
    const btnConfig = inject<DollarChusho | null>('$chusho', null)?.options
      ?.components?.btn;
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    let tag: any = 'button';
    let extraAttrs: Record<string, unknown> = {};

    if (this.to) {
      tag = resolveComponent('router-link');

      if (tag === 'router-link') {
        warn(
          `CBtn got a “to” prop but RouterLink component couldn’t be found. Is Vue Router installed? To make a CBtn behave like a traditional link, use the prop “href” instead.`
        );
      }

      extraAttrs = {
        to: this.to,
      };
    } else if (this.href) {
      tag = 'a';
      extraAttrs = {
        href: this.href,
      };
    }

    const elementProps: Record<string, unknown> = {
      ...extraAttrs,
      // Concerns only on button tags, skip for anchors
      ...(this.disabled && tag === 'button' && { disabled: true }),
      ...(this.type && tag === 'button' && { type: this.type }),
      ...generateConfigClass(btnConfig?.class, this.$props),
    };

    return h(tag, mergeProps(this.$attrs, elementProps), this.$slots);
  },
});
