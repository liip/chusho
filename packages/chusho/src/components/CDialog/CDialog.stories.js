import CDialog from './CDialog';
import { CBtn } from '../CBtn';

export default {
  title: 'Dialog',
  component: CDialog,
};

export const Default = () => ({
  components: { CDialog, CBtn },

  data() {
    return {
      open: false,
    };
  },

  template: `
    <div>
      <CBtn variant="default medium" @click="open = true">Open dialog</CBtn>

      <CDialog v-model="open" class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3" aria-labelledby="dialog-title">
        <header class="flex pb-3 mb-6 border-b">
          <div class="flex-1">
            <h1 id="dialog-title" class="text-lg font-semibold">Dialog title</h1>
          </div>
          <div class="ml-4">
            <CBtn @click="open = false" class="p-2 leading-none" aria-label="Close dialog">
              <span>✗</span>
            </CBtn>
          </div>
        </header>

        <p>
          Lorem ipsum dolor sit amet.
        </p>
      </CDialog>
    </div>
  `,
});

export const Nested = () => ({
  components: { CDialog, CBtn },

  data() {
    return {
      open: false,
      childOpen: false,
    };
  },

  template: `
    <div>
      <CBtn variant="default medium" @click="open = true">Open dialog</CBtn>

      <CDialog v-model="open" class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3" aria-labelledby="dialog-title">
        <header class="flex pb-3 mb-6 border-b">
          <div class="flex-1">
            <h1 id="dialog-title" class="text-lg font-semibold">Dialog title</h1>
          </div>
          <div class="ml-4">
            <CBtn @click="open = false" class="p-2 leading-none" aria-label="Close dialog">
              <span>✗</span>
            </CBtn>
          </div>
        </header>

        <p>
          Lorem ipsum dolor sit amet.
        </p>

        <div class="mt-4">
          <CBtn variant="default medium" @click="childOpen = true">Open Child Dialog</CBtn>
        </div>
      </CDialog>

      <CDialog v-model="childOpen" class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3" aria-labelledby="dialog-title">
        <div>
          Hello, I’m a nested Dialog, nice to meet you.
        </div>

        <CBtn class="font-semibold mt-4" @click="childOpen = false">Cancel</CBtn>
      </CDialog>
    </div>
  `,
});

export const OpenByDefault = () => ({
  components: { CDialog, CBtn },

  data() {
    return {
      open: true,
    };
  },

  template: `
    <div>
      <CBtn variant="default medium" @click="open = true">Open dialog</CBtn>

      <CDialog v-model="open" class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3" aria-labelledby="dialog-title">
        <header class="flex pb-3 mb-6 border-b">
          <div class="flex-1">
            <h1 id="dialog-title" class="text-lg font-semibold">Dialog title</h1>
          </div>
          <div class="ml-4">
            <CBtn @click="open = false" class="p-2 leading-none" aria-label="Close dialog">
              <span>✗</span>
            </CBtn>
          </div>
        </header>

        <p>
          Lorem ipsum dolor sit amet.
        </p>
      </CDialog>
    </div>
  `,
});

export const WithTransition = () => ({
  components: { CDialog, CBtn },

  data() {
    return {
      open: false,
    };
  },

  template: `
    <div>
      <CBtn variant="default medium" @click="open = true">Open dialog</CBtn>

      <CDialog v-model="open" :transition="{ name: 'fade' }" class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3" aria-labelledby="dialog-title">
        <header class="flex pb-3 mb-6 border-b">
          <div class="flex-1">
            <h1 id="dialog-title" class="text-lg font-semibold">Dialog title</h1>
          </div>
          <div class="ml-4">
            <CBtn @click="open = false" class="p-2 leading-none" aria-label="Close dialog">
              <span>✗</span>
            </CBtn>
          </div>
        </header>

        <p>
          Lorem ipsum dolor sit amet.
        </p>
      </CDialog>
    </div>
  `,
});
