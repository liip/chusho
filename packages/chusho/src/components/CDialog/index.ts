import CDialog from './CDialog';

export { CDialog };
