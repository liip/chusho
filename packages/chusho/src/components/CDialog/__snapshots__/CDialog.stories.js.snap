// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Dialog Default 1`] = `
<div>
  <button
    class="inline-block bg-blue-200 text-blue-900 rounded py-2 px-4"
    type="button"
  >
    Open dialog
  </button>
   
  <div
    aria-labelledby="dialog-title"
    class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3 v-portal"
    style="display: none;"
  />
</div>
`;

exports[`Storyshots Dialog Nested 1`] = `
<div>
  <button
    class="inline-block bg-blue-200 text-blue-900 rounded py-2 px-4"
    type="button"
  >
    Open dialog
  </button>
   
  <div
    aria-labelledby="dialog-title"
    class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3 v-portal"
    style="display: none;"
  />
   
  <div
    aria-labelledby="dialog-title"
    class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3 v-portal"
    style="display: none;"
  />
</div>
`;

exports[`Storyshots Dialog Open By Default 1`] = `
<div>
  <button
    class="inline-block bg-blue-200 text-blue-900 rounded py-2 px-4"
    type="button"
  >
    Open dialog
  </button>
   
  <div
    aria-labelledby="dialog-title"
    class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3 v-portal"
    style="display: none;"
  />
</div>
`;

exports[`Storyshots Dialog With Transition 1`] = `
<div>
  <button
    class="inline-block bg-blue-200 text-blue-900 rounded py-2 px-4"
    type="button"
  >
    Open dialog
  </button>
   
  <div
    aria-labelledby="dialog-title"
    class="w-full sm:w-3/4 md:w-1/2 lg:w-1/3 v-portal"
    style="display: none;"
  />
</div>
`;
